File: paiza/ruby/C/C057.rb
# 自分の得意な言語で
# Let's チャレンジ！！

ipt = gets.chomp.split(' ')
N = ipt[0].to_i
x = ipt[1].to_i
y = ipt[2].to_i

max_x = x

N.times do |i|
    
    dxdy = gets.chomp.split(' ')
    dx = dxdy[0].to_i
    dy = dxdy[1].to_i
    
    x += dx
    
    x > max_x ? max_x = x : max_x
    
    y += dy
    
    y <= 0 ? break : next
end

p max_x
    

File: paiza/php/C057.php
<?php
    // 自分の得意な言語で
    // Let's チャレンジ！！

    $ipt = explode(' ', fgets(STDIN));
    
    $x = $ipt[1];
    $y = $ipt[2];
    $max_x = $x;
    
    
    for($i; $i < $ipt[0]; $i++){
        $dxdy = explode(' ', fgets(STDIN));

        $dx = $dxdy[0];
        $dy = $dxdy[1];
        
        $x += $dx;
        if($x > $max_x){
            $max_x = $x;
        }
        
        $y += $dy;
        if($y <= 0){
            break;
        }
    }
    
    echo($max_x);
    
?>